spec/controllers/user/teams_controller_spec.rb

```diff
@@ -0,0 +1,63 @@
+require 'rails_helper'
+require 'user/teams_controller'
+
+describe User::TeamsController, type: :controller do
+  subject(:user) { create(:user, :with_team) }
+
+  describe "GET #index" do
+    context "when user is logged out" do
+      before { get :show }
+      it "redirects to sign in page" do
+        expect(response).to redirect_to(new_user_session_url)
+      end
+    end
+
+    context "when user is logged in" do
+      before do
+        sign_in user
+        get :show
+      end
+
+      it "responds with 200" do
+        expect(response).to have_http_status(:ok)
+      end
+    end
+  end
+
+  describe "PUT #update" do
+    context "when user is logged out" do
+      before { put :update }
+      it "redirects to sign in page" do
+        expect(response).to redirect_to(new_user_session_url)
+      end
+    end
+
+    context "when user is logged in" do
+      before do
+        sign_in user
+        team = create(:team, name: 'Foo Bar')
+        user.enrollments << create(:enrollment, team: team, is_admin: false)
+        put :update, team_slug: "foo-bar"
+      end
+
+      context "with an user's team slug" do
+        it "set current team slug session" do
+          expect(controller.session[:current_team_slug]).to eq('foo-bar')
+        end
+      end
+
+      # context "with a non user's team slug" do
```

Comment:
Is it supposed to be commented? If this should be ignored (if so, tell why in a comment) you can use `xcontext` so the `context` will be ignored by RSpec :)